crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -644,7 +644,14 @@ impl MoveTestAdapter<'_> for SuiTestAdapter {
 
                 let mut output = vec![];
                 if show_headers {
-                    output.push(format!("Headers: {:#?}", resp.http_headers.unwrap()));
+                    let mut headers_map: BTreeMap<_, _> = BTreeMap::new();
+
+                    for (header, value) in resp.http_headers.unwrap_or_default().into_iter() {
+                        if let Some(header) = header {
+                            headers_map.insert(format!("{header}"), value);
+                        }
+                    }
+                    output.push(format!("Headers: {:#?}", headers_map));
```

Comment:
nit (optional): Potentially slightly neater?

```suggestion
                    let headers_map: BTreeMap<_, _> = resp
                        .http_headers
                        .into_iter()
                        .flatten()
                        .filter_map(|(h, v)| Some((h?.to_string(), v)))
                        .collect();

                    output.push(format!("Headers: {headers_map:#?}"));
```

Reply:
Could you also update the same logic that handles header printing for `RunJsonRpc`, below? (Search for `Headers:` in this file).

Reply:
Done.